freqtrade/exchange/exchange.py

```diff
@@ -1572,6 +1642,7 @@ async def _async_fetch_trades(self, pair: str,
                     '(' + arrow.get(since // 1000).isoformat() + ') ' if since is not None else ''
                 )
                 trades = await self._api_async.fetch_trades(pair, since=since, limit=1000)
+            trades = self._trades_contracts_to_amount(trades)
```

Comment:
I'm ... hesitant about this (i'm actually not sure it's relevant to have the conversation for this method.).

This function is only called when using `freqtrade download-data --dl-trades` - which should only be called on kraken (all other supported exchanges provide OHLCV data directly).

Reply:
Taking it out just seems like putting in work to make it more difficult  should this ever be needed. What if another exchange ends up needing it or we add kraken futures, or someone else wants to add kraken futures, because I probably won't. If we don't need it, it will keep the sizes as they are

Reply:
kraken futures are a long way out i think - as ccxt does not support these either (it's a completely separate API - afaik).

if we don't need it - it's a call that's pointlessly consuming cpu cycles.

Let's leave it for now - I've noted it down as something to "look at again" before merging to develop.

Reply:
> which should only be called on kraken (all other supported exchanges provide OHLCV data directly).

Could this method of downloading OHLCV data be used for huobi?

Reply:
@samgermain no, huobi does not provide any historic data beyond 1500 (or 2000?) datapoints.

Their endpoint is called `Most recent trades` [huobi docs](https://huobiapi.github.io/docs/spot/v1/en/#get-the-most-recent-trades), and works similarly to the OHLCV endpoint.
Neither of the 2 even accept a "since/from/to" argument, only limit and symbol.
As such, you can only get the last `limit` datapoints - but never the data from "3 months ago".

Huobi itself will (probably) work for trading - but downloading historic data (and hence backtesting) will be impossible due to lack of data.
as such, it's an exchange we will not be able to recommend anytime soon.

to support huobi - huobi will need to change their API and/or add end points to allow downloading historic data. (i'm pretty sure that's also a problem in some areas for ccxt itself - and may raise quite some questions/issues on your side - unless almost noone is properly using that exchange).

If you have a contact there/are interested in having huobi supported by freqtrade, best get in touch with huobi / huobi support on adding this part.